libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -822,33 +829,54 @@ public void queryAsString(StringBuilder resultBuilder, QuerySpec querySpec, int
 			Cursor cursor = null;
 			try {
 				cursor = DBHelper.getInstance(db).limitRawQuery(db, sql, limit, querySpec.getArgs());
-				resultBuilder.append("[");
+
```

Comment:
@wmathurin  I wondering if we need to use StringBuffer here instead of StringBuilder? I do see that you are synchronizing on DB object. Maybe that is enough.

Reply:
StringBuffer has the same methods as the StringBuilder , but each method in StringBuffer is synchronized that is StringBuffer is thread safe. We don't need to synchronize here - so no reason to use the slower class.